src/java.base/share/classes/java/security/DigestInputStream.java

```diff
@@ -51,6 +51,15 @@
  * retain a handle onto the digest object, and clone it for each
  * digest to be computed, leaving the original digest untouched.
  *
+ * @implNote This implementation updates and only updates the message digest
+ *      with data actually read from the input stream when it is turned on.
+ *      This includes the various {@code read} methods, {@code transferTo},
+ *      {@code readAllBytes}, and {@code readNBytes}. Especially, data
```

Comment:
Maybe just "Note" instead of "Especially" (or just remove) as this doesn't seem to be an extra special situation.